kalite/main/migrations/0039_log_timestamps.py

```diff
@@ -0,0 +1,259 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+
+class Migration(DataMigration):
+
+    def forwards(self, orm):
+        """
+        Create latest_activity_timestamp data when possible by inspecting the model.
+        For ExerciseLog inspect relevant AttemptLogs to infer most recent activity.
+        If not available, and for other Logs, fallback to completion_timestamp if available.
+        All attempted migrations are wrapped in try-except blocks so that this migration
+        does not render an installation unusable simply because this fails for any data.
+        """
+        for e in orm.ExerciseLog.objects.filter(latest_activity_timestamp=None):
+
+            try:
+
+                alogs = orm.AttemptLog.objects.filter(user=e.user, exercise_id=e.exercise_id, context_type__in=["playlist", "exercise"])
+
+                if alogs:
+                    e.latest_activity_timestamp = sorted([a.timestamp for a in alogs])[-1]
+                    e.zone_fallback = e.get_zone()
```

Comment:
Oh, sorry, missed this in the first pass. This shouldn't be getting set here, if we're not running the overridden save method (so the zone stuff won't be getting affected, but changing this here might then break signatures).